### ISSUE TITLE
[7.7.0] Log NullReferenceException

### ISSUE THREAD

**Issue by BlazeGuy2** [StrangeLoopGames/EcoIssues#8923]
[output_log.txt](https://github.com/StrangeLoopGames/EcoIssues/files/2276041/output_log.txt)

```
(Filename: <33ac35150f0146e2a957fa0fbf850b3d> Line: 0)

Exception in view notify: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.NullReferenceException
  at (wrapper managed-to-native) UnityEngine.Animator.SetBoolString(UnityEngine.Animator,string,bool)
  at UnityEngine.Animator.SetBool (System.String name, System.Boolean value) [0x00001] in <0eccdc9deba94938961f5edcd68795a9>:0 
  at AvatarAnimationManager.UpdateTool () [0x00080] in <8d541fcbae7342d98b693432b5245e13>:0 
  at (wrapper managed-to-native) System.Reflection.MonoMethod.InternalInvoke(System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00032] in <f826c2584fc94ec19a48a6576640bdc5>:0 
   --- End of inner exception stack trace ---
  at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00048] in <f826c2584fc94ec19a48a6576640bdc5>:0 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in <f826c2584fc94ec19a48a6576640bdc5>:0 
  at Eco.Shared.View.Subscriber.Invoke () [0x00016] in <33ac35150f0146e2a957fa0fbf850b3d>:0 
  at Eco.Shared.View.View.NotifyChanged (System.String propname) [0x0007e] in <33ac35150f0146e2a957fa0fbf850b3d>:0 
```

**Comment by metachronism** [comment]
view callback exceptions should probably  be disabled in release, going to encounter a lot of these messages otherwise in the logs

**Comment by johnkslg** [comment]
Can we specifically try/catch for the 'object already GC'd' exception, so we still catch real exceptions and log them?

**Comment by metachronism** [comment]
null object ref is not the most common exception anyway, there are lots - and that would just be a bandaid solution.

the unsubscribe stuff is one way to resolve this.  If that code is only in master right now, that is fine, we can deal with one more release having potentially false "errors" in the log and consider this resolved for now.

**Comment by mirasrael** [comment]
@johnkslg @metachronism I'll do little investigation if it can be done w/o too much effort and dangerous changes. And If not then I'm agree with @metachronism that it can wait.

**Comment by mirasrael** [comment]
I did some investigation and the only more or less stable solution which I can see:
![image](https://user-images.githubusercontent.com/475445/44190562-05a4f900-a130-11e8-9612-6bde85f3c176.png)
When exception occurred it may be analyzed it it was `TargetInvocationException` and its `InnerException` is a `UnityEngine.MissingReferenceException` and it contains "has been destroyed".

I think it will work, but be aware about:
1. we need to remove this in master, because when we will start rewriting for Unsubscribe then we want to see all these places which still requires unsubscribing.
2. in theory this error may be produced in other cases - if we are using destroyed object in alive component in example, but we are not able to find these cases because of this spam anyway...

**Comment by metachronism** [comment]
unsubscribe should fix this.  Moving this off 7.7 release for now

**Comment by johnkslg** [comment]
Cool, looks like you can detect the majority of cases. You could make it release-only that those errors are hidden. And as Eric says perhaps it will no longer matter later.

**Comment by mirasrael** [comment]
Decided to not fix it in 7.7 and instead provide proper fix with Unsubscribe in next release.
